esp/services/ws_workunits/ws_workunitsHelpers.hpp

```diff
@@ -641,5 +646,67 @@ class CWsWuEmailHelper
 
     void send(const char *body, const void *attachment, size32_t lenAttachment, StringArray &warnings);
 };
+
+class CGetThorSlaveLogToFileThreadParam : public CInterface
+{
+    WsWuInfo* wuInfo;
+    unsigned slaveNum;
+    StringAttr groupName, logDate, logDir, fileName;
+
+public:
+    IMPLEMENT_IINTERFACE;
+
+    CGetThorSlaveLogToFileThreadParam(WsWuInfo* _wuInfo, const char*_groupName,
+        const char*_logDate, const char*_logDir, unsigned _slaveNum, const char* _fileName)
+        : wuInfo(_wuInfo), groupName(_groupName), logDate(_logDate), logDir(_logDir),
+          slaveNum(_slaveNum), fileName(_fileName) { };
+
+    virtual void doWork()
+    {
+        MemoryBuffer dummy;
+        wuInfo->getWorkunitThorSlaveLog(groupName.get(), nullptr, logDate.get(),
+            logDir.get(), slaveNum, dummy, fileName.get(), false);;
+    }
+};
+
+class CGetThorSlaveLogToFileThread : public CSimpleInterfaceOf<IPooledThread>
+{
+public:
+    virtual void init(void* _param) override
+    {
+        param.setown((CGetThorSlaveLogToFileThreadParam*)_param);
+    }
+    virtual void threadmain() override
+    {
+        param->doWork();
+        param.clear();
+    }
+
+    virtual bool canReuse() const override
+    {
+        return true;
+    }
+    virtual bool stop() override
+    {
+        return true;
+    }
+   
+private:
+    Owned<CGetThorSlaveLogToFileThreadParam> param;
+};
+
+//---------------------------------------------------------------------------------------------
+
+class CGetThorSlaveLogToFileThreadFactory : public CInterface, public IThreadFactory
```

Comment:
can probably be a CSimpleInterfaceOf<IThreadFactory>

Reply:
Will change.